### ISSUE TITLE
[Hacktoberfest] Wiki/Documentation

### ISSUE THREAD

**Issue by tr7zw** [tr7zw/Item-NBT-API#191]
Hey there.
Sadly the wiki(as the intended primary documentation of code examples/how-to-s) has been quite unmaintained for a long time. Some examples are only available on the horrible SpigotMC's page [sourcecode](https://github.com/tr7zw/Item-NBT-API/blob/master/spigot.bb), the Spigot forums, or Discord.
Therefore, I set up the wiki's source to be auto-deployed from the master branch of this repository(https://github.com/tr7zw/Item-NBT-API/tree/master/wiki). From now on, code examples, pitfalls, guides, and documentation can be supplied to the wiki with a normal pull request. 
For Hacktoberfest, feel free to create a piece of example code, clean up the wording, or update the documentation of the wiki. Then provide it as a valid pull request for Hacktoberfest.

Some examples that need updated code examples/guides:
- Creating player heads with textures in 1.8-1.19
- Keeping custom data from an item -> placed BlockEntity -> back to item
- Adding custom logic to items(on right-click/drop/pickup/other events)
- Adding custom data to 1.8-1.13 mobs via some item slot
- Adding custom data to 1.14+ Entities and BlockEntities
- Copy over data from one item to another
- Keeping track of custom Entities without causing lag
- Keeping track of custom player properties without causing lag
- Storing an inventory in an item/non inventory block entity
- Shading on Gradle
- Links to the places where the API can be downloaded(Modrinth/Bukkit)
- Better link to the Javadoc(https://tr7zw.github.io/Item-NBT-API/v2-api/ https://tr7zw.github.io/Item-NBT-API/v2-plugin/)
- Documentation of plugins/servers? using the NBTAPI(instead of SpigotMCs description)

**Comment by c99SRS** [comment]
HI @tr7zw can you assign this one to me please?

**Comment by tr7zw** [comment]
> HI @tr7zw can you assign this one to me please?

The issue is open in general since I don't expect anyone to do ALL of that. For Hacktoberfest afaik only the PRs are important(tagged correctly).

**Comment by kangarko** [comment]
@tr7zw AI models have gotten way better now, you can just dump the plugin's code into gemini 2.5 pro with 1M context window or claude 3.7 sonnet max, from my experience it is pretty reliable in generating docs if you check everything after it. Also tip, you can download the entire wiki as one folder and open it in cursor: https://github.com/tr7zw/Item-NBT-API.wiki.git

**Comment by tr7zw** [comment]
The wiki is already in the repo itself. Also ai is useless to create meaningful documentation in this case since it's missing too much context.

**Comment by kangarko** [comment]
Yeah the context issue was big in the past, but now with 1M context window it is largely solved. I recommend you give Cursor a try and load up all java classes to the chat while on Gemini, I was really surprised myself. It's only out for a week or so.

**Comment by tr7zw** [comment]
It's not about context size. Without the knowledge of the mc internals/changes of the past 10 years, implementation differences between Spigot/Paper and what people actually use the API for, AI can't produce anything meaningful that can't be read from method names and existing Javadoc already.

**Comment by broken1arrow** [comment]
I will say you should only feed so much you can easy check and make adjustments self and also ask the AI change stuff. Just batch in 1 million context will just make it hard and time consuming fix all mistakes. 

I self mostly using it as rubber duck and just using it for things that is just monotonous to do self.  Just relay on the AI does not work, as you often need fix things self and take more time than just doing it self (not always but often).
